<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>902</width>
    <height>689</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="MplCanvas" name="raw" native="true">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="processed" native="true">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QWidget" name="widget" native="true">
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>320</height>
         </size>
        </property>
        <widget class="QTableWidget" name="tableWidget">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>0</y>
           <width>270</width>
           <height>270</height>
          </rect>
         </property>
         <property name="maximumSize">
          <size>
           <width>270</width>
           <height>270</height>
          </size>
         </property>
        </widget>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>270</y>
           <width>111</width>
           <height>41</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Kernel size</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
        <widget class="QSpinBox" name="spinBox">
         <property name="geometry">
          <rect>
           <x>210</x>
           <y>270</y>
           <width>81</width>
           <height>41</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>3</number>
         </property>
         <property name="maximum">
          <number>11</number>
         </property>
         <property name="singleStep">
          <number>2</number>
         </property>
        </widget>
        <widget class="QRadioButton" name="medianBtn">
         <property name="geometry">
          <rect>
           <x>350</x>
           <y>60</y>
           <width>86</width>
           <height>20</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Median</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
        <widget class="QRadioButton" name="maxBtn">
         <property name="geometry">
          <rect>
           <x>350</x>
           <y>100</y>
           <width>86</width>
           <height>20</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Max</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
        <widget class="QRadioButton" name="minBtn">
         <property name="geometry">
          <rect>
           <x>350</x>
           <y>140</y>
           <width>86</width>
           <height>20</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Min</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
        <widget class="QRadioButton" name="gaBtn">
         <property name="geometry">
          <rect>
           <x>350</x>
           <y>180</y>
           <width>111</width>
           <height>20</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Gaussian</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
        <widget class="QRadioButton" name="laBtn">
         <property name="geometry">
          <rect>
           <x>350</x>
           <y>220</y>
           <width>111</width>
           <height>20</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Laplacian</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
        <widget class="QPushButton" name="startBtn">
         <property name="geometry">
          <rect>
           <x>590</x>
           <y>120</y>
           <width>151</width>
           <height>51</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Start process</string>
         </property>
        </widget>
        <widget class="QPushButton" name="resetBtn">
         <property name="geometry">
          <rect>
           <x>590</x>
           <y>60</y>
           <width>151</width>
           <height>51</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Reset all</string>
         </property>
        </widget>
        <widget class="QPushButton" name="randomBtn">
         <property name="geometry">
          <rect>
           <x>330</x>
           <y>270</y>
           <width>151</width>
           <height>41</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Random kernel</string>
         </property>
        </widget>
        <widget class="QRadioButton" name="customBtn">
         <property name="geometry">
          <rect>
           <x>350</x>
           <y>20</y>
           <width>111</width>
           <height>20</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>20</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Custom</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">buttonGroup</string>
         </attribute>
        </widget>
        <widget class="QSpinBox" name="medianBox">
         <property name="geometry">
          <rect>
           <x>470</x>
           <y>60</y>
           <width>61</width>
           <height>21</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>3</number>
         </property>
         <property name="maximum">
          <number>11</number>
         </property>
         <property name="singleStep">
          <number>2</number>
         </property>
        </widget>
        <widget class="QSpinBox" name="maxBox">
         <property name="geometry">
          <rect>
           <x>470</x>
           <y>100</y>
           <width>61</width>
           <height>21</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>3</number>
         </property>
         <property name="maximum">
          <number>11</number>
         </property>
         <property name="singleStep">
          <number>2</number>
         </property>
        </widget>
        <widget class="QSpinBox" name="minBox">
         <property name="geometry">
          <rect>
           <x>470</x>
           <y>140</y>
           <width>61</width>
           <height>21</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>3</number>
         </property>
         <property name="maximum">
          <number>11</number>
         </property>
         <property name="singleStep">
          <number>2</number>
         </property>
        </widget>
        <widget class="QSpinBox" name="gaBox">
         <property name="geometry">
          <rect>
           <x>470</x>
           <y>180</y>
           <width>61</width>
           <height>21</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <pointsize>16</pointsize>
          </font>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="minimum">
          <number>3</number>
         </property>
         <property name="maximum">
          <number>11</number>
         </property>
         <property name="singleStep">
          <number>2</number>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MplCanvas</class>
   <extends>QWidget</extends>
   <header>mplcanvas.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
